RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,17 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    edm::LogError("PhotonProducer")
+      << "Error! Can't get the preshowerEcalHits";
+    validPreshowerRecHits=false;
+  }
```

Comment:
My request was for the module to be internally consistent. Either they all throw with missing inputs or none of them do: 
https://github.com/kmcdermo/cmssw/blob/c40a39d42fb57dd84be7b6646c110610053f3178/RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc#L194-L214